tests/e2e/suite_test.go

```diff
@@ -33,24 +33,57 @@ import (
 )
 
 const (
-	samplesDir = "../../config/samples/"
+	samplesDir = "../../config/samples"
+	testConf   = "./conf"
 )
 
+const (
+	e2eNamespaceDrill    = "kubestone-e2e-drill"
+	e2eNamespaceFio      = "kubestone-e2e-fio"
+	e2eNamespaceIperf3   = "kubestone-e2e-iperf3"
+	e2eNamespacePgbench  = "kubestone-e2e-pgbench"
+	e2eNamespaceSysbench = "kubestone-e2e-sysbench"
+)
+
+var e2eNamespaces = []string{
+	e2eNamespaceDrill,
+	e2eNamespaceFio,
+	e2eNamespaceIperf3,
+	e2eNamespacePgbench,
+	e2eNamespaceSysbench,
+}
+
 var restClientConfig = ctrl.GetConfigOrDie()
 var client ctrlclient.Client
 var ctx = context.Background()
 var scheme = runtime.NewScheme()
 
-func init() {
+var _ = BeforeSuite(func() {
 	_ = k8sscheme.AddToScheme(scheme)
 	_ = perfv1alpha1.AddToScheme(scheme)
 
 	var err error
 	client, err = ctrlclient.New(restClientConfig, ctrlclient.Options{Scheme: scheme})
 	if err != nil {
-		panic(err)
+		Fail(err.Error())
 	}
-}
+
+	for _, namespace := range e2eNamespaces {
+		_, _, err = run("kubectl create namespace " + namespace)
+		if err != nil {
+			Fail(err.Error())
+		}
+	}
+})
+
+var _ = AfterSuite(func() {
```

Comment:
given that all the tests are running in parallel we will have timing issues in the future.
I'd recommend limiting the number of parallel tests to 2-4 AND let the test do the cleanup so that
resources are freed after each test run. Will create a separate ticket for that.

Reply:
tracked here: https://github.com/xridge/kubestone/issues/134